google-cloud-translate/MIGRATING.md

```diff
@@ -0,0 +1,248 @@
+ *  Previously, some V3 client classes included class methods for constructing
+    resource paths. These paths are now instance methods on the client objects,
+    and are also available in a separate paths module. See
+    [Resource Path Helpers](#resource-path-helpers) for more info.
+ *  Some classes have moved into different namespaces. See
+    [Class Namespaces](#class-namespaces) for more info.
+
+### Library Structure
+
+Older releases of the `google-cloud-translate` gem were all-in-one gems that
+included potentially multiple clients for multiple versions of the Translation
+service. The `Google::Cloud::Translate.new` factory method would
+return you an instance of a `Google::Cloud::Translate::V2::Api`
+object for the V2 version of the service, or a
+`Google::Cloud::Translate::V3::TranslationServiceClient` object for the
+V3 version of the service. All these classes were defined in the same gem.
+
+With the 3.0 release, the `google-cloud-translate` gem still provides factory
+methods for obtaining clients. (The method signatures will have changed. See
+[Creating Clients](#creating-clients) for details.) However, the actual client
+classes have been moved into separate gems, one per service version. The
+`Google::Cloud::Translate::V2::Api` class, along with its
+helpers and data types, is now part of the `google-cloud-translate-v2` gem.
+Similarly, the `Google::Cloud::Translate::V3::TranslationService::Client`
+class is part of the `google-cloud-translate-v3` gem. 
+
+For normal usage, you can continue to install the `google-cloud-translate` gem
+(which will bring in the versioned client gems as dependencies) and continue to
+use factory methods to create clients. However, you may alternatively choose to
+install only one of the versioned gems. For example, if you know you will use only
+`V2` of the service, you can install `google-cloud-translate-v2` by itself, and
+call `Google::Cloud::Translate::V2.new` to create V2 clients directly.
+
+### Creating Clients
+
+In older releases, to create a client object, you would use the
+`Google::Cloud::Translate.new` class method. Keyword arguments were available to
+select a service version and to configure parameters such as credentials and
+timeouts. Furthermore, you could configure default parameters using the
+`Google::Cloud::Translate.configure` method.
+
+In the 3.0 release, there are separate class methods for creating clients of the
+modern (V3) and legacy (V2) Translation services. To create a V2 client, use the
+`translation_v2_service` class method, which takes the same keyword arguments
+you would have used previously. To create a V3 (or later) client, use the
+`translation_service` class method and set options in a configuration block
+rather than passing keyword arguments.
+
+Old:
+```
+client = Google::Cloud::Translate.new credentials: "/path/to/credentials.json"
+```
+
+New (V2):
+```
+client = Google::Cloud::Translate.translation_v2_service(
```

Comment:
Could this also be written as `Google::Cloud::Translate.translation_service version: :v2`?

Reply:
No it cannot, because the method signature is different. The V2 method still takes keyword arguments, whereas V3 (and all subsequent versions) follow our new pattern of a configuration block.

Reply:
This is a tough situation, but I like how you updated it with the version number in the example title.